tabbycat/motions/statistics.py

```diff
@@ -160,25 +139,25 @@ def _prefetch_prelim_motions(self):
         it's used by someone with BP."""
 
         self.prelim_motions = Motion.objects.filter(
-            round__tournament=self.tournament,
-            round__stage=Round.STAGE_PRELIMINARY,
-            round__debate__ballotsubmission__confirmed=True,
-        ).order_by('round__seq').select_related('round')
+            tournament=self.tournament,
+            rounds__stage=Round.STAGE_PRELIMINARY,
+            rounds__debate__ballotsubmission__confirmed=True
+        ).prefetch_related('rounds').order_by('rounds__seq', 'roundmotions__seq')
 
         annotations = {}  # dict of keyword arguments to pass to .annotate()
-        annotations['ndebates'] = Count('round__debate__ballotsubmission', distinct=True)
+        annotations['ndebates'] = Count('ballotsubmission', distinct=True)
 
         annotations.update({'%s_average' % side: Avg(
-            'round__debate__ballotsubmission__teamscore__points',
-            filter=Q(round__debate__ballotsubmission__teamscore__debate_team__side=side),
+            'ballotsubmission__teamscore__points',
+            filter=Q(rounds__debate__ballotsubmission__teamscore__debate_team__side=side),
```

Comment:
So, this gets a little complicated. You noticed below that you needed to drop the `round__debate__` from the filter keyword arguments in order to get it to filter for the right things. Turns out the same is true in this line, where—at first glance—it should read
``` python
filter=Q(ballotsubmission__teamscore__debate_team__side=side)
```

You might then wonder, why did we go through that roundabout `round___debate__ballotsubmission__` path, when we could just look relevant ballots up directly from `ballotsubmission__`? The reason is that, in formats where everyone in a round does the same motion, we didn't want the lack of record of motion in a ballot to inhibit motion statistics. That is: if we know that everyone in a round is doing the same motion, we really just want statistics from the _round_, not the motion. So the motion statistics page looks for statistics for the round.

At this point the filters in the annotations can interact with the main filter up on line 141, which can make things tricky. It's also not entirely clear to me what behavior we'd even want—are we continuing the assumption that there is a one-to-one relationship between motion and round, in the BP code? If so, I think using `rounds__debate__ballotsubmission__` everywhere works?

Reply:
That goes with the difference between motion-round statistics and motion statistics. I've fixed it so that motion-round statistics take precedence, and pure motion statistics take from the whole set though. Fixed (and seems like the code is simplified a bit)